### PR TITLE
Project: Authentication - Members only

### PR DESCRIPTION
**Pull request by VanessaAoki**
### This Pull request contains the following features:
- Update ReadMe file
- Created a user with the username, email, and password before;
- Create an anonymous post 
- Read posts from the feed
- Edited and destroyed posts

#### Note: Sign in or sign up before create any post

### CODE REVIEW

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Invalid :red_circle: 
Do not make any changes in config files (.github/workflows/linters.yml) - they represent style guidelines that you share with your team - which is a group of all Microverse students.

That's it, you can submit another code review request after making these changes :recycle: :innocent:

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
+  devise_for :users, controllers: { registrations: 'registrations' }
+  resources :posts
```
Comment (whiz25) [inline config/routes.rb:5]: - Change this to: `resources :posts, only: [:new, :create, :index]`

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Nice work on this project. Your design is amazing and most minimum requirements have been met.
- A user can authenticate, create a post 
- A user can only see post when signed in
Well done.

## Changes
- Display post content for guest users. Right now the landing page displays the login form.
- Check my comments for suggestions

### Martin Afani
#### Slack: whiz25

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,16 +1,67 @@
...
+  		<div class="notification is-primary global-notification">
+  			<p class="notice"><%= notice %></p>
+  		</div>
+  	<% end %>
+
+  	<% if flash[:alert] %>
+  		<div class="notification is-danger global-notification">
+  			<p class="alert"><%= alert %></p>
+  		</div>
+  	<% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb:22]: There should be absolutely NO logic in views.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,16 +1,67 @@
...
+                  <%= link_to "Logout", destroy_user_session_path, method: :delete, class:"button is-info" %>
+                </p>
+              <% else %>
+              <p class="control">
+                <%= link_to 'Sign In', new_user_session_path, class: "button is-info" %>
+              </p>
+              <p class="control">
+                <%= link_to 'Sign Up', new_user_registration_path, class: "button is-info" %>
+              </p>
+              <% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb:57]: There should be absolutely NO logic in views.

File: app/views/posts/_feed.html.erb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  			<div class="content">
+  				<p><%= post.post %></p>
+  			</div>
+        <% if user_signed_in? && current_user.id == post.user_id %>
+  			<nav class="level">
+  				<div class="level-left is-mobile">
+						
+  					<%= link_to post, class: "level-item" do %>
+  					  <span class="icon"><i class="fa fa-link" aria-hidden="true"></i></span>
+  					<% end %>
```
Comment (acushlakoncept) [inline app/views/posts/_feed.html.erb:33]: There should be absolutely NO logic in views.

File: app/views/posts/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    </div>
+  </div>
+  
+  <%= f.error_notification message: 'Please, fullfil all the requirements' %>
+  <% if user_signed_in? %>
+    <%= f.button :submit, class: "button is-info is-outlined" %>
+  <% else %>
+    <button class="button is-danger" title="Disabled button" disabled>You must be loged in to post</button>
+  <% end %>
+<% end %>
```
Comment (acushlakoncept) [inline app/views/posts/_form.html.erb:15]: There should be absolutely NO logic in views.

File: app/views/posts/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<section class="section">
+  <div class="container">
+    <div class="columns">
+    	<% if user_signed_in? %>
+    		<%= render 'profile' %>
+      <% end %>
```
Comment (acushlakoncept) [inline app/views/posts/index.html.erb:6]: There should be absolutely NO logic in views.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hello @VanessaAoki and @Emmyn5600, Great job! 👏  Both of you have done well on this project.

You've done an awesome job so far 👌. 
✅  Kudos on the extra step to deploy your app even though its not required, 👏 👏 
✔️  I love your UI, it's a impressive project you've got here 👏 

It's a minor change.

- [ ] I ran into an error, running `rails db:migrate` caused by this file: `20210426203220_drop_tweets.rb`
- [ ] As a guest you can see only post's text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). Well done on using Avatar, you should disable user avatar for guest user too and allow only text, since my identity can be reveal as seen on the screenshot 
Screen Shot 2021-04-27 at 7 39 54 PM


### Consider the requested changes below as OPTIONAL, since the previous TSE may have missed it. If you can try to implement it before submitting another code review.

- [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Comment by Emmyn5600** [comment]
Thanks, Uduak Essien for your advice as well as the shared link you sent to us, to use a helper folder for the logic implementation instead of the view folder. We promise you that we will change that to the next project.

**Review by GzimAsani — APPROVED** [review]
# STATUS: Project Accepted :green_circle: :partying_face: 

## Hello **Team** great job implementing all the changes mentioned by previous TSE-s. You can merge this pull request and good luck with the next projects :+1: 
### However I would strongly reccomend you removing the logic from the views because it is very important to learn how to put all your logic in your helpers and use them in your views


# HAPPY CODING :computer: :desktop_computer:

**Pull request merged**